ribes_utils.py

```diff
@@ -0,0 +1,70 @@
+import itertools
+from collections import defaultdict
+import math
+
+def count_ngram(sent, order=2):
+  gram_pos = dict()
+  for i in range(order):
+    gram_pos[i+1] = defaultdict(lambda: [])
+
+  for i, word in enumerate(sent):
+    for j in range(min(i+1, order)):
+      gram_pos[j+1][word].append(i-j)
+      word = sent[i-j-1] + ' ' + word
+  
+  return gram_pos
+
+def word_alignment(ref, out, order=2):
```

Comment:
I think this is good because it basically resolves https://github.com/neulab/compare-mt/issues/17

Maybe we could split this function off into a file `align_utils` and have it be `ngram_context_align` or something? Then you could have a comment indicating that this is the algorithm proposed by RIBES. Then `RibesScorer` would call this function to generate alignments, but we could also use it for other things, such as the word accuracy calculations (which are using a different implicit alignment algorithm).